storybook/main.js

```diff
@@ -35,22 +26,52 @@ export default {
         '@storybook/addon-a11y',
         // "@storybook/addon-storysource"
     ],
-    nodeResolve: {
-        exportConditions: ['browser', 'development'],
+    framework: {
+        name: 'storybook-web-components-wds',
     },
-
-    rollupConfig(config) {
+    wdsFinal(config) {
+        const json = fromRollup(rollupJson);
+        return mergeConfigs(config, {
+            nodeResolve: {
+                exportConditions: ['browser', 'development'],
+                moduleDirectories: [
+                    'node_modules',
+                    'packages',
+                    'projects',
+                    'tools',
+                ],
+            },
+            clearTerminalOnReload: false,
+            watch: true,
+            mimeTypes: {
+                '**/*.json': 'js',
+            },
+            plugins: [json(), watchSWC()],
+            http2: true,
+            middleware: [
+                async (ctx, next) => {
+                    await next();
+                    if (
+                        // Icon packages
+                        ctx.url.search('/icons-') > -1 ||
+                        // Node modules
+                        (ctx.url.search('/node_modules/') > -1 &&
+                            ctx.url.search('/@spectrum-web-components') === -1)
+                    ) {
+                        ctx.set(
+                            'Cache-Control',
+                            'public, max-age=604800, stale-while-revalidate=86400'
```

Comment:
the flakyness it partially related to this caching, so DevTools helped me because I had "Disable cache" active in Network tab

but the root of the issue is this https://github.com/modernweb-dev/web/pull/2365